drizzlepac/haputils/quality_analysis.py

```diff
@@ -425,9 +446,257 @@ def generate_plots(json_data):
                      plotname=resids_name)
         fig_id += 1
 
+# -------------------------------------------------------------------------------
+# Generate the Bokeh plot for the pipeline astrometric data.
+#
+HOVER_COLUMNS = ['gen_info.instrument',
+                 'gen_info.detector',
+                 'gen_info.filter',
+                 'gen_info.imgname',
+                 'header.DATE-OBS',
+                 'header.RA_TARG',
+                 'header.DEC_TARG',
+                 'header.GYROMODE',
+                 'header.EXPTIME',
+                 'fit_results.aligned_to']
+                   
+TOOLTIPS_LIST = ['INSTRUMENT', 'DET', 'FILTER', 
+                  'EXPNAME', 'DATE', 'RA', 'DEC', 'GYRO', 'EXPTIME',
+                  'ALIGNED_TO']
+INSTRUMENT_COLUMN = 'full_instrument'
+
+RESULTS_COLUMNS = ['fit_results.rms_x',
+                   'fit_results.rms_y',
+                   'fit_results.xsh',
+                   'fit_results.ysh',
+                   'fit_results.rot',
+                   'fit_results.scale',
+                   'fit_results.rot_fit',
+                   'fit_results.scale_fit',
+                   'fit_results.nmatches',
+                   'fit_results.skew']
+
+SOURCE_COLUMNS = ['residuals.x',
+                  'residuals.y',
+                  'residuals.ref_x',
+                  'residuals.ref_y']
+TOOLSEP_START = '{'
+TOOLSEP_END = '}'
+DETECTOR_LEGEND = {'UVIS': 'magenta', 'IR': 'red', 'WFC': 'blue', 
+                    'SBC': 'yellow', 'HRC': 'black'}
+    
 
-                     
+def build_tooltips(tips):
+    """Return list of tuples for tooltips to use in hover tool.
+    
+    Parameters
+    ----------
+    tips : list
+        List of indices for the HOVER_COLUMNS entries to be used as tooltips 
+        to be included in the hover tool.
+
+    """
+    tools = [(TOOLTIPS_LIST[i], '@{}{}{}'.format(
+                                TOOLSEP_START, 
+                                HOVER_COLUMNS[i],
+                                TOOLSEP_END)) for i in tips]
+    
+    return tools
+    
+def get_pandas_data(csv_filename):
+    """Load the harvested data, stored in a CSV file, into local arrays.
+
+    Parameters
+    ==========
+    csv_filename: str
+    Name of the CSV file created by the harvester.
+
+    Returns
+    =======
+    phot_data: Pandas dataframe
+    Dataframe which is a subset of the input Pandas dataframe written out as
+    a CSV file.  The subset dataframe consists of only the requested columns
+    and rows where all of the requested columns did not contain NaNs.
+
+    (mean_dMagAp1_mean, mean_dMagAp1_median): tuple
+    Aperture 1 mean of means and mean of medians
+
+    (mean_dMagAp2_mean, mean_dMagAp2_median): tuple
+    Aperture 2 mean of means and mean of medians
+    """
```

Comment:
These comments belong to the photometry_graphics.py and not here.

Reply:
Comments were removed in 7ba7497.